mgs1600gy_control/src/mgs1600gy_sensor.cpp

```diff
@@ -0,0 +1,282 @@
+// Copyright 2023 HarvestX Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mgs1600gy_control/mgs1600gy_sensor.hpp"
+
+namespace mgs1600gy_control
+{
+Mgs1600gySensor::~Mgs1600gySensor()
+{
+  boost::interprocess::shared_memory_object::remove(this->shm_->get_name());
+  this->shm_.reset();
+}
+
+CallbackReturn Mgs1600gySensor::on_init(const hardware_interface::HardwareInfo & info)
+{
+  if (hardware_interface::SensorInterface::on_init(info) != CallbackReturn::SUCCESS) {
+    return CallbackReturn::ERROR;
+  }
+
+  SENSOR_MIN_ = std::stof(info.hardware_parameters.at("sensor_min"));
+  SENSOR_MAX_ = std::stof(info.hardware_parameters.at("sensor_max"));
+  NAME_ = info.hardware_parameters.at("name");
+  BASE_LINK_ = NAME_ + "_link";
+  MAGNET_LINK_ = NAME_ + "_magnet_link";
+
+  // TODO : set parameters
```

Comment:
mgs1600gy_node gets parameters by using param.yaml which is written in launch file.
But this new harware file can't get parameters such as "roll" or "orientation_covariance" through param.yaml.
So this file needs to be added to the changes to set up parameters.

Reply:
I've got it.
I should be written in the controller.yaml files.